pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -430,8 +430,8 @@ private SuccessResponse addSchema(Schema schema, boolean override, boolean force
       throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.BAD_REQUEST, e);
     } catch (Exception e) {
       _controllerMetrics.addMeteredGlobalValue(ControllerMeter.CONTROLLER_SCHEMA_UPLOAD_ERROR, 1L);
-      throw new ControllerApplicationException(LOGGER, String.format("Failed to add new schema %s.", schemaName),
-          Response.Status.INTERNAL_SERVER_ERROR, e);
+      throw new ControllerApplicationException(LOGGER, String.format("Failed to add new schema %s. Reason: %s",
```

Comment:
Is there a format that we followed for such error logs?
I like this format as it will make easier to search/ grep.

Reply:
I vote against this messages created with String.format. They are difficult to read and may affect performance (in a negative way). In this case the performance is not problematic, but it is just a bad idea to try to do that everywhere

Reply:
As a developer who learnt the trade on C using `printf` and `snprintf`, this is the most natural templatization method for me :) 

It is unfortunate that the Java implementation is terrible. A benchmark that shows the terrible performance of `String.format`. Source: https://stackoverflow.com/a/1281651 

`StringBuilder` or `+` are possible candidates. I find `StringBuilder` more natural because `+` to me are for numbers and it confuses my brain for a new nanoseconds.

Reply:
> Is there a format that we followed for such error logs?

I see this format being followed at other places where a custom message is followed by `Reason: <exception message>` hence extending the same here.


>I vote against this messages created with String.format

Agree, in this PR I am not touching the existing formatted messages, just adding the reason part where its missing. Cleaning up all the `String.format` can be a separate PR altogether.

Reply:
opened an issue to discuss and finalize which option to use https://github.com/apache/pinot/issues/14404